File: cpp/1110.cpp
#include <iostream>
#include <vector>

using namespace std;

void stack(int n)
{
    vector<int> a, b;
    int i;

    for (i = 1; i <= n; i++) {
        a.push_back(i);
    }

    while (n > 1) {
        b.push_back(a.at(0));
        a.erase(a.begin());
        a.push_back(a.at(0));
        a.erase(a.begin());
        n--;
    }

    n = b.size();
    n--;
    cout << "Discarded cards: ";

    for (i = 0; i < n; i++) {
        cout << b.at(i) << ", ";
    }

    cout << b.at(i) << "\n";
    cout << "Remaining card: " << a.at(0) << '\n';
}

int main()
{
    int n;

    while (true) {
        cin >> n;
        if (n == 0) break;
        stack(n);
    }

    return 0;
}

File: py/1110.py
def stack(n):
    a = [i for i in range(1, n + 1)]
    b = []

    while len(a) > 1:
        b.append(a.pop(0))
        a.append(a[0])
        a.pop(0)

    print('Discarded cards:', end=' ')
    print(*b, sep=', ')
    print('Remaining card:', a[0])


while True:
    n = int(input())

    if n == 0:
        break

    stack(n)